### PR TITLE
Minor: Document how to test for trailing whitespace in `slt` / sqllogictests

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

- Follow on to https://github.com/apache/datafusion/pull/13197

## Rationale for this change

@findepi  figured out a good way to test for trailing spaces, let's put that in the user documentation as suggested by https://github.com/apache/datafusion/pull/13197#discussion_r1824888498

## What changes are included in this PR?

Add docs and slt examples

## Are these changes tested?

Yes
## Are there any user-facing changes?

docs and tests, no functional changes

### CODE REVIEW

File: datafusion/sqllogictest/README.md [hunk 1]
```diff
@@ -102,6 +102,39 @@ SELECT * from foo;
 
 Assuming it looks good, check it in!
 
+## Cookbook: Testing for whitespace
+
+The `sqllogictest` runner will automatically strip trailing whitespace, meaning
+it additional effort to verify that trailing whitespace is correctly produced
```
Comment (findepi) [inline datafusion/sqllogictest/README.md:108]:
```suggestion
The `sqllogictest` runner will automatically strip trailing whitespace, meaning
it requires an additional effort to verify that trailing whitespace is correctly produced
```

File: datafusion/sqllogictest/README.md [hunk 2]
```diff
@@ -102,6 +102,39 @@ SELECT * from foo;
...
+
+```text
+query T
+select substr('Andrew Lamb', 1, 7)
+----
+Andrew
+```
+
+To test trailing whitespace, project additional non-whitespace column on the
+right. For example, by selecting `'XX'` after the column of interest, the test
```
Comment (findepi) [inline datafusion/sqllogictest/README.md:121]: It's any non-whitespace, but we can take opportunity to suggest what we want to see in tests.
What about `'|'` that's used in one test i added, or `'$'` (not non-typical for marking end of line)?

**Review by findepi — APPROVED** [review]
thanks!

File: datafusion/sqllogictest/README.md (hunk 2 above)
Comment (alamb) [reply datafusion/sqllogictest/README.md:121]: Let's use `|` to be consistent with what you added . I will update the PR

**Review by jonathanc-n — APPROVED** [review]
This lgtm!

**Review by jonahgao — APPROVED** [review]

**Pull request merged**